src/jira.py

```diff
@@ -0,0 +1,161 @@
+        """
+        super().__init__()  # Initiate the super class, Issue
+
+        self.url = get_access_params('jira')['options']['server'] % (default_orgs['jira'])
+        self.token = self.token = get_access_params('jira')['api_token']
+        self.headers = {'Authorization': 'Basic ' + self.token}
+
+        if key:
+            r = requests.get("%ssearch?jql=id=%s" % (self.url, key), headers=self.headers).json()
+
+            if "issues" in r.keys():  # If the key doesn't match any issues, this will be an empty list
+                response = r["issues"][0]  # Get the one and only issue in the response
+            else:
+                raise ValueError("No issue matching this id was found")
+
+        if response["fields"]["assignee"]:
+            self.assignee = response["fields"]["assignee"]["name"]
+        self.created = datetime.datetime.strptime(response["fields"]["created"].split('.')[0], '%Y-%m-%dT%H:%M:%S')
+        self.description = response["fields"]["description"]
+        self.issue_type = response["fields"]["issuetype"]["name"]
+        self.jira_key = response["key"]
+        self.parent = response["fields"]["customfield_10008"]  # This custom field holds the epic link
+        self.status = response["fields"]["status"]["name"]
+        self.summary = response["fields"]["summary"]
+        self.updated = datetime.datetime.strptime(response["fields"]["updated"].split('.')[0], '%Y-%m-%dT%H:%M:%S')
+
+        # Not all issue descriptions have the corresponding github issue listed in them
+        self.github_repo_name, self.github_key = self.get_github_equivalent() or (None, None)
+
+        if "customfield_10014" in response["fields"].keys():  # This custom field holds story point values
+            self.story_points = response["fields"]["customfield_10014"]
+
+    def get_github_equivalent(self):
+        """Find the equivalent Github issue key and repo name if listed in the issue text. Issues synced by unito-bot
+        will have this information."""
+
+        if self.description:
+            match_obj = re.search(r'Repository Name: ([\w_-]*).*\n.*Issue Number: ([\w-]*)', self.description)
+            if match_obj:
+                return match_obj.group(1), match_obj.group(2)
+            print("No match was found in the description.")
+
+    def dict_format(self) -> dict:
+        """Describe this issue in a dictionary that can be posted to Jira"""
+
+        d = {
+            "fields": {  # these fields can be updated
+                "assignee": {"name": self.assignee},
+                "description": self.description,
+                "summary": self.summary
+            }
+        }
+
+        if self.story_points:
+            d["fields"]["customfield_10014"] = self.story_points
+
+        return d
+
+    def update_remote(self):
+        """Update the remote issue. The issue must already exist in Jira."""
+
+        transitions = {  # Jira API uses these codes to identify status changes
```

Comment:
This is great. I would put into a separate file (e.g., `transitions.json`) or into `settings.py` for now.

Reply:
moved to settings.py